parsing/pprintast.ml

```diff
@@ -608,7 +610,7 @@ and core_type1 ctxt f x =
     | Ptyp_open(li, ct) ->
        pp f "@[<hov2>%a.(%a)@]" longident_loc li (core_type ctxt) ct
     | Ptyp_extension e -> extension ctxt f e
-    | (Ptyp_arrow _ | Ptyp_alias _ | Ptyp_poly _) ->
+    | (Ptyp_arrow _ | Ptyp_alias _ | Ptyp_poly _ | Ptyp_of_kind _) ->
```

Comment:
The `paren true` here makes me think this case will cause double parens.

Reply:
I do not understand why my comment above is wrong, but it evidently is. I've added a test. Moving on.